tests/test_misc.py

```diff
@@ -648,6 +648,61 @@ def test_multiplexed_rpc(node_factory):
         assert obj['id'] == l1.rpc.decoder.decode(i.decode("UTF-8"))['id']
     sock.close()
 
+
+def test_malformed_rpc(node_factory):
+    """Test that we get a correct response to malformed RPC commands"""
+    l1 = node_factory.get_node()
+
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.connect(l1.rpc.socket_path)
+
+    commands = [
+        # No id.
+        # No method
+        b'{"id":1,"jsonrpc":"2.0","method":"getinfo","params":[]}',
+        b'{"id":1,"jsonrpc":"2.0","params":[2000]}',
+        # Bad ID
+        b'{"id":{2},"jsonrpc":"2.0","method":"dev-slowcmd","params":[1500]}',
+        b'{"id":2,"jsonrpc":"2.0","method":"dev-slowcmd","params":[1500]}',
+        b'{"id":3,"jsonrpc":"2.0","method":"dev-slowcmd","params":[1000]}',
+        b'{"id":3,"jsonrpc":"2.0","method":"dev-slowcmd","params":[1000]}',
+        b'{"id":4,"jsonrpc":"2.0","method":"dev-slowcmd","params":[500]}',
+        b'{"id":4,"jsonrpc":"2.0","method":"dev-slowcmd","params":[500]}'
+    ]
+    # no ID
```

Comment:
The comments seem to be misaligned with the actual tests.

Reply:
This looks like a good usecase for table driven testing: create a dict with the inputs as keys and the expected outcome as the values and then iterate through them.